index.js

```diff
@@ -38,10 +39,16 @@ function Store (dbName, options) {
   var db = new CustomPouchDB(dbName)
   var emitter = new EventEmitter()
   var remote = options.remote
+  var HoodieSync = db.hoodieSync({remote: remote, emitter: emitter})
   var api = merge(
-    db.hoodieSync({remote: remote, emitter: emitter}),
     db.hoodieApi({emitter: emitter}),
-    { hasLocalChanges: hasLocalChanges.bind(db) }
+    HoodieSync,
+    {
+      hasLocalChanges: hasLocalChanges.bind(db),
+      push: syncWrapper.bind(HoodieSync, 'push'),
+      pull: syncWrapper.bind(HoodieSync, 'pull'),
+      sync: syncWrapper.bind(HoodieSync, 'sync')
```

Comment:
I think what I'd do is something like this

```
var syncApi = db.hoodieSync({remote: remote})
var api = merge(
  db.hoodieApi({emitter: emitter}),
  {
    hasLocalChanges: hasLocalChanges.bind(db),
    push: syncWrapper.bind(syncApi, 'push'),
    pull: syncWrapper.bind(syncApi, 'pull'),
    sync: syncWrapper.bind(syncApi, 'sync')
  }
)
subscribeToSyncEvents(syncApi, emitter)
```

Don't pass emitter to `db.hoodieSync`, let it have it's own. Then have a method subscribeToSyncEvents where we subscribe to the events emitted in `syncAp`, normalise the passed objects, and re-trigger the events with our `emitter` instance that we also pass to `db.hoodieApi({emitter: emitter})`

Does that make sense?